### ISSUE TITLE
Autodoc logo should be configurable

### ISSUE THREAD

**Issue by linusg** [ziglang/zig#19279]
### Zig Version

0.12.0-dev.3245+4f782d1e8

### Steps to Reproduce and Observed Behavior

As much as I love Zig, with autodoc being a generic part of the build API the Zig logo should not be hardcoded in the HTML file :^)

### Expected Behavior

Some way to specify a custom logo.

**Comment by andrewrk** [comment]
So, I was thinking that customization of generated docs is important, and often the way to do that would be providing an alternate style sheet, or something like that right? Well, my thought process was that index.html is only 415 lines and that includes the css, logo, and layout too. So you have a high degree of flexibility by providing your own index.html file, right? I imagine it's more than just the logo that would be nice to customize.

**Comment by linusg** [comment]
> So you have a high degree of flexibility by providing your own index.html file, right? I imagine it's more than just the logo that would be nice to customize.

I agree that would be nice for maximum flexibility, but at the same time it's a much higher barrier to having project-themed API docs so I'd much prefer to (also) have an easier option.

> What alternate interface are you envisioning for this customization process, as opposed to replacing `index.html`?

In the Python world I've used sphinx before which allows you to specify a logo [like this](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_logo):

> **html_logo**
> 
> If given, this must be the name of an image file (path relative to the [configuration directory](https://www.sphinx-doc.org/en/master/glossary.html#term-configuration-directory)) that is the logo of the docs, or URL that points an image file for the logo. It is placed at the top of the sidebar; its width should therefore not exceed 200 pixels. Default: None.
> 
> _New in version 0.4.1_: The image file will be copied to the `_static` directory of the output HTML, but only if the file does not already exist there.
> 
> _Changed in version 4.0_: Also accepts the URL for the logo file.

Also relevant is the option to [add directories for additional static files](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_static_path):

> **html_static_path**
> 
> A list of paths that contain custom static files (such as style sheets or script files). Relative paths are taken as relative to the configuration directory. They are copied to the output’s _static directory after the theme’s static files, so a file named default.css will overwrite the theme’s default.css.
> 
> As these files are not meant to be built, they are automatically excluded from source files.
> 
> _Changed in version 0.4_: The paths in [html_static_path](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_static_path) can now contain subdirectories.
> 
> _Changed in version 1.0_: The entries in [html_static_path](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_static_path) can now be single files.
> 
> _Changed in version 1.8_: The files under [html_static_path](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_static_path) are excluded from source files.

In zig it could be an options struct passed to `getEmittedDocs()`:

```
docs/
  index.html
  static/
    logo.png
```

```
// build.zig
compile.getEmittedDocs(.{
    .title = "My Project",
    .logo = .{ .path = "docs/static/logo.png" },
});
```

Or something like this for the index.html-replacing way:

```
// build.zig
compile.getEmittedDocs(.{
    .index_html = .{ .path = "docs/index.html" },
    .extra_static_files = .{ .path = "docs/static" }, // index.html can now reference static/{logo.png,styles.css,script.js,...}
});
```

**Comment by andrewrk** [comment]
Being able to easily configure generated autodocs from a build script is a good use case. I see the appeal of having this be part of the Compile step rather than some kind of tacked-on patch-up.

Hmm, it would be nice to keep index.html static though and not have to do template substitution... :thinking: 

On the other hand, maybe template substitution is not so bad after all.

**Comment by andrewrk** [comment]
Another thing that should be configurable is that we want to add `<link rel="search" type="application/opensearchdescription+xml" title="Zig Standard Library Documentation" href="search.xml">` to the zig std lib docs but clearly that title needs to be customized. However my earlier comments still apply, you can just maintain a different index file and overwrite it as part of the deployment process. That solves this problem too.

Anyway, let's start by providing the maximum flexibility option, and then solving the other use cases can be a follow-up topic. I like your proposed build system API.
